### PR TITLE
Fix: add touchscreen tests

### PR DESCRIPTION
**Pull request by stanley31huang**
validate the required touch events for touchscreen also add new test to cover maximum fingers support on touchscreen

### CODE REVIEW

**Comment by stanley31huang** [comment]
#### Validate checkbox provider
https://pastebin.canonical.com/p/hcqQfhJgvV/

#### side-load checkbox results
- 4 fingers: https://certification.canonical.com/hardware/202212-30991/submission/327102/
- 10 fingers: https://certification.canonical.com/hardware/202212-30991/submission/327101/

**Comment by stanley31huang** [comment]
Per talked with @patliuu , attached the reference links for touch screen in Python scripts.

File: checkbox-provider-ce-oem/units/touchscreen/jobs.pxu [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+template-filter: device.category == 'TOUCHSCREEN'
+template-engine: jinja2
+template-unit: job
+plugin: user-interact
+category_id: com.canonical.plainbox::touchscreen
+id: ce-oem-touchscreen/evdev/single-touch-tap-{{ product_slug }}
+imports: from com.canonical.certification import device
+estimated_duration: 10.0
+_description:
+  PURPOSE:
```
Comment (LiaoU3) [inline checkbox-provider-ce-oem/units/touchscreen/jobs.pxu:12]: What do you think if we write it like `_purpose`, `_steps` or  `_verification`? Cause this kind of style is supported in checkbox.

**Review by LiaoU3 — CHANGES_REQUESTED** [review]

**Comment by stanley31huang** [comment]
#### Side-loading test results:
```
avnet@ubuntu:~$ checkbox-reading.checkbox-cli
Using sideloaded provider: checkbox-provider-ce-oem, version 0.1 from /var/tmp/checkbox-providers/checkbox-provider-ce-oem
=========================[ Resume Incomplete Session ]==========================
There are 0 incomplete sessions that might be resumed
Preparing...
============[ Bootstrap com.canonical.certification::device (1/1) ]=============
Reports will be saved to: /home/avnet/snap/checkbox-reading/7/.local/share/checkbox-ng
==============[ Running job 1 / 2. Estimated time left: 0:00:20 ]===============
[ ce-oem-touchscreen/evdev/single-touch-tap-Silicon_Integrated_System_Co._SiS_HID_Touch_Controller ]
ID: com.canonical.qa.ceoem::ce-oem-touchscreen/evdev/single-touch-tap-Silicon_Integrated_System_Co._SiS_HID_Touch_Controller
Category: com.canonical.plainbox::touchscreen
Purpose:

Validate that single-touch tap is properly detected

Steps:

1. Commence the test
   2. Tap the screen with one finger.

Pick an action
    => press ENTER to continue
  c => add a comment
  s => skip this job
  q => save the session and quit
[csq]:
... 8< -------------------------------------------------------------------------

### Available input devices:
- Integrated Camera: Integrated C
- Silicon Integrated System Co. SiS HID Touch Controller

### Start to capture events from Silicon Integrated System Co. SiS HID Touch Controller
time 1692604614.057028 type 3 (EV_ABS), code 47   (ABS_MT_SLOT), value 0
time 1692604614.057028 type 3 (EV_ABS), code 57   (ABS_MT_TRACKING_ID), value 96
time 1692604614.057028 type 3 (EV_ABS), code 53   (ABS_MT_POSITION_X), value 1878
time 1692604614.057028 type 3 (EV_ABS), code 54   (ABS_MT_POSITION_Y), value 2554
time 1692604614.057028 type 1 (EV_KEY), code 330  (BTN_TOUCH), value 1
time 1692604614.057028 type 3 (EV_ABS), code 0    (ABS_X), value 1878
time 1692604614.057028 type 3 (EV_ABS), code 1    (ABS_Y), value 2554

#### 1 touch event detected seccessfully
### Stop to capture events from Silicon Integrated System Co. SiS HID Touch Controller
------------------------------------------------------------------------- >8 ---
Outcome: job passed
==============[ Running job 2 / 2. Estimated time left: 0:00:10 ]===============
[ ce-oem-touchscreen/evdev/maximum-touch-tap-Silicon_Integrated_System_Co._SiS_HID_Touch_Controller ]
ID: com.canonical.qa.ceoem::ce-oem-touchscreen/evdev/maximum-touch-tap-Silicon_Integrated_System_Co._SiS_HID_Touch_Controller
Category: com.canonical.plainbox::touchscreen
Purpose:

Validate that maximum-touch tap is properly detected
the maximum finger support is 10 by default
please modify the TOUCHSCREEN_MAX_FINGERS variable

Steps:

1. Commence the test
2. Tap the screen with 10 fingers
(Or the maximum supported fingers for this touchscreen)

Pick an action
    => press ENTER to continue
  c => add a comment
  s => skip this job
  q => save the session and quit
[csq]:
... 8< -------------------------------------------------------------------------
Tap the screen with 4 fingers simultaneously.


### Available input devices:
- Integrated Camera: Integrated C
- Silicon Integrated System Co. SiS HID Touch Controller

### Start to capture events from Silicon Integrated System Co. SiS HID Touch Controller

#### Touch events are not expected
time 1692604621.028708 type 3 (EV_ABS), code 57   (ABS_MT_TRACKING_ID), value 97
time 1692604621.028708 type 3 (EV_ABS), code 53   (ABS_MT_POSITION_X), value 755
time 1692604621.028708 type 3 (EV_ABS), code 54   (ABS_MT_POSITION_Y), value 2612
time 1692604621.028708 type 3 (EV_ABS), code 47   (ABS_MT_SLOT), value 1
time 1692604621.028708 type 3 (EV_ABS), code 57   (ABS_MT_TRACKING_ID), value 98
time 1692604621.028708 type 3 (EV_ABS), code 53   (ABS_MT_POSITION_X), value 1574
time 1692604621.028708 type 3 (EV_ABS), code 54   (ABS_MT_POSITION_Y), value 2257
time 1692604621.028708 type 3 (EV_ABS), code 47   (ABS_MT_SLOT), value 2
time 1692604621.028708 type 3 (EV_ABS), code 57   (ABS_MT_TRACKING_ID), value 99
time 1692604621.028708 type 3 (EV_ABS), code 53   (ABS_MT_POSITION_X), value 3515
time 1692604621.028708 type 3 (EV_ABS), code 54   (ABS_MT_POSITION_Y), value 2802
time 1692604621.028708 type 1 (EV_KEY), code 330  (BTN_TOUCH), value 1
time 1692604621.028708 type 3 (EV_ABS), code 0    (ABS_X), value 755
time 1692604621.028708 type 3 (EV_ABS), code 1    (ABS_Y), value 2612

#### Touch events are not expected
time 1692604621.035687 type 3 (EV_ABS), code 47   (ABS_MT_SLOT), value 0
time 1692604621.035687 type 3 (EV_ABS), code 53   (ABS_MT_POSITION_X), value 754
time 1692604621.035687 type 3 (EV_ABS), code 54   (ABS_MT_POSITION_Y), value 2613
time 1692604621.035687 type 3 (EV_ABS), code 47   (ABS_MT_SLOT), value 1
time 1692604621.035687 type 3 (EV_ABS), code 53   (ABS_MT_POSITION_X), value 1573
time 1692604621.035687 type 3 (EV_ABS), code 0    (ABS_X), value 754
time 1692604621.035687 type 3 (EV_ABS), code 1    (ABS_Y), value 2613
time 1692604621.042627 type 3 (EV_ABS), code 47   (ABS_MT_SLOT), value 2
time 1692604621.042627 type 3 (EV_ABS), code 54   (ABS_MT_POSITION_Y), value 2803
time 1692604621.042627 type 3 (EV_ABS), code 47   (ABS_MT_SLOT), value 3
time 1692604621.042627 type 3 (EV_ABS), code 57   (ABS_MT_TRACKING_ID), value 100
time 1692604621.042627 type 3 (EV_ABS), code 53   (ABS_MT_POSITION_X), value 2302
time 1692604621.042627 type 3 (EV_ABS), code 54   (ABS_MT_POSITION_Y), value 2382

#### 4 touch event detected seccessfully
### Stop to capture events from Silicon Integrated System Co. SiS HID Touch Controller
------------------------------------------------------------------------- >8 ---
Outcome: job passed
WARNING:checkbox-ng.launcher.stages:Using side-loaded providers disabled the certification report
 ☑ : ce-oem-touchscreen/evdev/single-touch-tap-Silicon_Integrated_System_Co._SiS_HID_Touch_Controller
 ☑ : ce-oem-touchscreen/evdev/maximum-touch-tap-Silicon_Integrated_System_Co._SiS_HID_Touch_Controller
```

**Review by LiaoU3 — APPROVED** [review]
LGTM

File: checkbox-provider-ce-oem/bin/evdev_touch_test.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+                )
+                break
+            events = []
+        else:
+            events.append(event)
+    print("### Stop to capture events from {}".format(d.name))
+
+
+if __name__ == "__main__":
+    global events
```
Comment (patliuu) [inline checkbox-provider-ce-oem/bin/evdev_touch_test.py:224]: Thanks for adding the comments based on our offline talk.
Regarding this global variable, do we still need it?

File: checkbox-provider-ce-oem/units/touchscreen/test-plan.pxu [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+id: after-suspend-ce-oem-touchscreen-evdev
+unit: test plan
+_name: Touchscreen evdev tests (after suspend)
+_description:
+  Touchscreen evdev tests
+include:
+  after-suspend-ce-oem-touchscreen/evdev/single-touch-tap-.*
+  after-suspend-ce-oem-touchscreen/evdev/maximum-touch-tap-.*
+bootstrap_include:
+  com.canonical.certification::device
```
Comment (patliuu) [inline checkbox-provider-ce-oem/units/touchscreen/test-plan.pxu:21]: Should we have 4 spaces indent also for test plans?

Comment (stanley31huang) [reply checkbox-provider-ce-oem/units/touchscreen/test-plan.pxu]: I have pushed a commit for this MR, thanks.

File: checkbox-provider-ce-oem/bin/evdev_touch_test.py (hunk 2 above)
Comment (stanley31huang) [reply checkbox-provider-ce-oem/bin/evdev_touch_test.py]: Nope, we don't need this global events variable
I have pushed a commit for this MR, thanks.

**Review by patliuu — APPROVED** [review]
LGTM. Thank you
